docker/zenml-server-hf-spaces.Dockerfile

```diff
@@ -0,0 +1,96 @@
+ARG PYTHON_VERSION=3.9
+FROM python:${PYTHON_VERSION}-slim AS base
```

Comment:
I'm wondering, can we simply replace this by something like:

```
ARG PYTHON_VERSION=3.9
ARG ZENML_VERSION

FROM zenmldocker/zenml-server:{ZENML_VERSION:+==$ZENML_VERSION}-py{$PYTHON_VERSION}
```

to save duplicating the entire file?

Reply:
ok let me try that. Most of the other spaces don't use an `ENTRYPOINT` whereas we do, so I wanted to allow people to pass in their custom ENV variables.

Reply:
I updated it as sparsely as seemed to work. It still requires overwriting the `ENTRYPOINT` and `CMD` instructions, but indeed much better this way.